opencti-platform/opencti-front/src/components/dataGrid/dataTableHooks.ts

```diff
@@ -0,0 +1,29 @@
+import { useFragment } from 'react-relay';
+import type { GraphQLTaggedNode, OperationType } from 'relay-runtime';
+import type { KeyType } from 'react-relay/relay-hooks/helpers';
+import { DataTableColumn, DataTableVariant, UseDataTable } from './dataTableTypes';
+import usePreloadedPaginationFragment, { UsePreloadedPaginationFragment } from '../../utils/hooks/usePreloadedPaginationFragment';
+import { useFormatter } from '../i18n';
+import useEntityToggle from '../../utils/hooks/useEntityToggle';
+import { computeLink } from '../../utils/Entity';
+import useLocalStorage, { UseLocalStorageHelpers } from '../../utils/hooks/useLocalStorage';
+
+export const useLineData = (lineFragment: GraphQLTaggedNode) => (row: KeyType) => useFragment(lineFragment, row);
```

Comment:
For proper typescript we should use generic types, like relay
```suggestion
export const useLineData = <TKey>(lineFragment: GraphQLTaggedNode) => (row: KeyType) => useFragment<TKey>(lineFragment, row);
```
this can be a gros chantier